odoo/fields.py

```diff
@@ -44,6 +50,9 @@
 from odoo.exceptions import CacheMiss
 from odoo.osv import expression
 
+_T = TypeVar("_T", bound=Any)
+_type = type
```

Comment:
```suggestion
T = TypeVar("T", bound=Any)
```
Just use `type` directly and a simpler version of type var.